gax-java/gax/src/test/java/com/google/api/gax/retrying/AbstractRetryingExecutorTest.java

```diff
@@ -222,29 +223,32 @@ public void testTotalTimeoutExceeded() throws Exception {
   }
 
   @Test
-  public void testCancelOuterFutureBeforeStart() throws Exception {
-    FailingCallable callable = new FailingCallable(4, "request", "SUCCESS", tracer);
+  public void testCancelOuterFutureBeforeStart() {
+    // Use MockCallable instead of FailingCallable as we do not need the request to be re-tried.
+    // For this test, the callable should attempt to run and then see that the external future
+    // is cancelled and that the callable's execution has ended.
+    MockCallable callable = new MockCallable(tracer, "request", "SUCCESS");
 
     RetrySettings retrySettings =
         FAST_RETRY_SETTINGS
             .toBuilder()
             .setInitialRetryDelay(Duration.ofMillis(1_000L))
             .setMaxRetryDelay(Duration.ofMillis(1_000L))
-            .setTotalTimeout(Duration.ofMillis(10_0000L))
+            .setTotalTimeout(Duration.ofMillis(10_000L))
             .build();
     RetryingExecutorWithContext<String> executor =
         getExecutor(getAlgorithm(retrySettings, 0, null));
     RetryingFuture<String> future = executor.createFuture(callable, retryingContext);
     boolean res = future.cancel(false);
-
     assertTrue(res);
+    assertTrue(future.isCancelled());
```

Comment:
I believe `assertFutureCancel()` is used to verify that the future was cancelled after the executorService has submitted the callable for execution and waits to ensure that it was cancelled properly. Since this was cancelled prior to being executed, we don't need to wait to verify that the future is cancelled.